src/components/eip4626/user/VaultDepositWithdraw.tsx

```diff
@@ -0,0 +1,199 @@
+import {
+  Box,
+  Text,
+  Flex,
+  Input,
+  Tabs,
+  TabList,
+  TabPanels,
+  Tab,
+  TabPanel,
+  Button,
+  Alert,
+  AlertDescription,
+  AlertIcon,
+} from "@chakra-ui/react";
+import Icon from "@/components/Icon";
+import { useMemo } from "react";
+import { useDepositWithdrawFlow } from "@/hooks/useDepositWithdrawFlow";
+import { EvmAddress } from "@/types/types";
+import { VaultActionResults } from "@/components/eip4626/user/VaultActionResults";
+
+type Props = {
+  vaultAddress: EvmAddress;
+  toggleError: (isError: boolean) => void;
+  error: boolean;
+};
+
+export const VaultDepositWithdraw = ({
+  vaultAddress,
+  toggleError,
+  error,
+}: Props) => {
+  const {
+    withdrawForm,
+    depositForm,
+    depositError,
+    approveError,
+    depositResult,
+    approveResult,
+    shareUserBalance,
+    vaultAssetUserBalance,
+    approveToken,
+  } = useDepositWithdrawFlow(vaultAddress);
+
+  const maxAmounts = useMemo(
+    () => ({
+      deposit: vaultAssetUserBalance
+        ? parseFloat(vaultAssetUserBalance?.toString())
+        : 0,
+      withdraw: shareUserBalance ? parseFloat(shareUserBalance?.toString()) : 0,
+    }),
+    [shareUserBalance, vaultAssetUserBalance],
+  );
+
+  const handleUpdateAmountWithPercent = (
+    percentage: number,
+    isDeposit = false,
+  ) => {
+    const calculatedAmount =
+      (maxAmounts[isDeposit ? "deposit" : "withdraw"] * percentage) / 100;
+    const form = isDeposit ? depositForm : withdrawForm;
+
+    if (calculatedAmount) {
+      form.setFieldValue("amount", calculatedAmount);
+
+      if (calculatedAmount <= maxAmounts[isDeposit ? "deposit" : "withdraw"]) {
+        toggleError(false);
+      } else {
+        toggleError(true);
+      }
+    }
+  };
+
+  const _renderInputSection = (isDeposit = false) => {
+    const form = isDeposit ? depositForm : withdrawForm;
```

Comment:
The approach of "internal" renders and conditional switching is not very flexible, we should have totally separated forms for depo and withdrawals, and then see how much of those are similar, then start separating out the common parts to make them reusable and shareable if needed.

Reply:
Having those "isDeposit" checks everywhere in this component brings excessive complexity, makes readability harder. Having separate components, even with a lot of duplicated functionality at the start, would be easier to improve in the long run.